pandas/tests/reshape/merge/test_merge.py

```diff
@@ -670,6 +673,8 @@ def test_join_append_timedeltas(self):
                 "t": [timedelta(0, 22500), timedelta(0, 22500)],
             }
         )
+        if using_array_manager:
+            expected = expected.astype(object)
```

Comment:
why this change?

Reply:
Similar as the comment above: because `df` has object dtype, and for now in this PR I am following the casting rules.

Will add a comment about it.